### ISSUE TITLE
Purchased servers not appearing in scan - analyze, only in scan, and no links to them.

### ISSUE THREAD

**Issue by DJ-Laser** [danielyxie/bitburner#3061]
When I buy servers, they don't appear in scan-analyze, but they do appear in scan. This is annoying because I have to constantly type out names of my servers, and there are no links to them, but I think If they were in scan-analyze it would just fill the list with pserver-1,2,3... and clog up the screen. Maybe it could say:
"And (number servers) of your servers" at the end of the list, that you could click to get a link list of the servers, or this could also be a new terminal command, or just make autolink.exe work for scan too. 

Steps to reproduce:

1. Buy some servers
2. Use scan:
![Screenshot 2022-03-02 07 41 24](https://user-images.githubusercontent.com/90146620/156363640-54a8ee6a-332b-4688-8a22-752efcc041ed.png)
3. Use scan-analyze:
![Screenshot 2022-03-02 07 42 08](https://user-images.githubusercontent.com/90146620/156363710-cedf77ec-08ee-40d9-828f-f51411993fdb.png)

TLDR: There should be links to purchased servers.

Version; Bitburner v1.4.0 (a3c599e6)

**Comment by Andres-M-Rios** [comment]
Well, correct me if I am wrong developers, as I am first and foremost and player of this excellent game, and I have no expertise in the choices made in the development of 'bitburner' but from my knowledge of networks and what I would do in today's internet with custom attack servers is hide them behind a firewall. The only way to see my attacks would be deep packet inspection on my otherwise 'bland' web surfing traffic from my home machine. (To hide as much as possible) In this vain, I feel like a surface scan to analyze (E.G. scan-analyze command) would not turn up my 'hidden' servers.

As such I fee this is not a bug, but a development decision.

If you really hate it feel free to 'fork' and add in your functionality to review.

I recommend looking into scripting your server. I am using a NS2 script to run my server farm, and it is very quick and effective.

`/** @param {NS} ns **/
export async function main(ns) {
	const server_prefix = "tower";
	var i = 1;
	while (i < 26 ) {
		var current_server = server_prefix.concat(i);
		ns.killall(current_server);
		ns.exec('hack.js', current_server, 426, 'comptek' );
		i++;
		ns.print(current_server);
	}
}`

That's all it takes to automate all your numbered servers to attack one target, in this case 'comptek' because my hack skills not high enough for some of the juicier targets on my internet in game.

I'm gonna mark this as not a bug, in my humble opinion.

**Comment by Andres-M-Rios** [comment]
Also, you can just run the command 'connect pserver-1' do something and log out, but I recommend learning scripting so you can batch whatever you want to all the servers in one command.
My script kills any script running on the server and deploys 'hack.js' to run attacking 'comptek' right now. You can set whatever target you want by changing the string to your target.

**Comment by phyzical** [comment]
@Andres-M-Rios i think what this user is requesting is for a way to visualize it via the command-line function.

tbh i don't see why we cant add them, i would guess the code is going "scan analyze any server we can backdoor" and well your not supposed to backdoor private servers so they probably don't get listed. (but ill check later today and see if i can see any other logic suggesting why)

**Comment by Andres-M-Rios** [comment]
Yes, if you want to make the change, I think the other reason this particular player was hoping to have this functionality was so that when 'create program link' is completed in the game, it will allow click access to one of the particular bought server just by running scan-analyze and clicking the link. I think that is the main goal here in his ticket.

**Comment by DJ-Laser** [comment]
I think there should be a thing to click to view the list, instead of automatically filling the list up with 25 servers ( could clog the he terminal full of servers you didn't want to look at ) I'll change the issue title since what I am looking for is just an easy way to autolink to them.

**Comment by DJ-Laser** [comment]
Ok so I looked into it and you can use the -a flag to see purchased servers. I did not know this before, but I still think autolink should work in scan as well as scan-analyze is this an ok change for a pull request or does autilink not work with scan for a reason?

**Comment by DJ-Laser** [comment]
I'll close this, as I didn't know functionality for this existed.

**Comment by Andres-M-Rios** [comment]
Makes sense, fully tested the -a flag and everything functionality wise works perfectly. Not a bug, just someone who didn’t read the full help… ;-)

Andes M. Partridge-Rios
***@***.***

**Comment by phyzical** [comment]
@DJ-Laser i think scan is a "simple" command but "scan-analyse" requires more hacking skill to make ect so i think its valid it doesnt autolink
